tests/baselines/reference/useUnknownInCatchVariables02(strict=false).errors.txt

```diff
@@ -0,0 +1,23 @@
+tests/cases/compiler/useUnknownInCatchVariables02.ts(4,10): error TS2339: Property 'name' does not exist on type '{}'.
+tests/cases/compiler/useUnknownInCatchVariables02.ts(11,10): error TS2339: Property 'name' does not exist on type '{}'.
+
+
+==== tests/cases/compiler/useUnknownInCatchVariables02.ts (2 errors) ====
+    try {
+        // ...
+    }
+    catch ({ name }) {
+             ~~~~
+!!! error TS2339: Property 'name' does not exist on type '{}'.

```

Comment:
This seems like a weird error -- where is `{}` coming from?

Reply:
@RyanCavanaugh I think there are two ways to throw the error, use `TS2339` directly in condition with `useUnknownInCatchVariables`. or use `getBindingElementTypeFromParentType` similarly to parameter binding element which actually has error reporting `TS2339` and uses any type instead of `unknown`.

```ts
function bar({ foo }: unknown) {
   foo // any
}
```

Reply:
I'd think the latter, so everything is consistent with destructuring `unknown` elsewhere. [Playground Link](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAbgLhgVzAazCA7mGBeGARgCYBmAFgG4AoW0SWAbxgDMQQYBffeGgej4whwmAD0A-EA)

Reply:
I just debugged this and realized that this is the `strict=false` test; the `strict=true` test shows the desired error (and the code is doing the "latter" I said above).

This seems to be because `getApparentType` returns the empty object for `unknown` when `strictNullChecks` is disabled.

Reply:
And further, `getPropertyTypeForIndexType` uses that reduced apparent type for its error messages, not the original type.

Reply:
Surprisingly, this is already the behavior, it's just that there's no test for it!

![image](https://user-images.githubusercontent.com/5341706/149591570-a8b93c40-c373-4e67-9f14-d77aed37066e.png)

Reply:
For fun, I removed the conversion of `unknown` to `{}`, and honestly, I can't say I'm a fan of other existing errors like:

```
 !!! error TS2322:  Type 'unknown' is not assignable to type 'A'.
 !!! error TS2322:    Type 'ReturnType<FunctionsObj<T>[string]>' is not assignable to type 'A'.
 !!! error TS2322:       Type 'unknown' is not assignable to type 'A'.
 !!! error TS2322:         Property 'x' is missing in type '{}' but required in type 'A'.
```

I mean, obviously we know that `unknown` isn't assignable to anything, so what's the point in picking out a prop of `A` to say that it's missing that prop?

Reply:
I filed #47494 to cover this.

Reply:
So, in the context of this PR, I don't think that this actually is anything new and is okay to go in; we can use the above issue to talk about changing the apparent type of `unknown` (which would help this and other cases, but maybe break things).